File: javascript/binarySearch.js
let array = [];
let n = Math.floor(Math.random() * 100);
for (let i = 0; i < n; i++) {
	array.push(Math.floor(Math.random() * 1000));
}
array.sort((a, b) => a - b);
let key = array[Math.floor(Math.random() * n)];

console.log("Initial Array: " + array + "\nElement to find: " + key);

var startTime = performance.now();

let p = 0;
let q = array.length - 1;
function binarySearch(A, p, q) {
	if (p > q) {
		return "not in the array";
	}
	let pos = Math.floor((p + q) / 2);
	if (A[pos] === key) {
		return pos;
	}
	if (A[pos] < key) {
		return binarySearch(A, pos + 1, q);
	}
	if (A[pos] > key) {
		return binarySearch(A, p, pos - 1);
	}
}

console.log("The key is in position: " + binarySearch(array, p, q));

var endTime = performance.now();
console.log("Execution Time: " + (endTime - startTime) + " ms!");

File: python/binarySearch.py
import time
import random
import math

def binarySearch(A, key, low, high):

	if low > high :
		return 

	m = math.floor( (low + high) / 2 )

	if A[m] == key:
		return m
	elif A[m] < key:
		return binarySearch(A, key, m+1, high)
	else:
		return binarySearch(A, key, low, m-1)


A = [random.randint(1, 100) for _ in range(10)]
A.sort()

print("Array: ", A)

key = int(input("Search an element: "))

low = 0
high = len(A) - 1

startTime = time.time()

index = binarySearch(A, key, low, high)

endTime = time.time()
executionTime =  endTime - startTime

if index is None:
	print("The element doesn't exist")
else:
	
	print("The element is located in", index, "position of the array.")

	

print(f"Execution time: {executionTime:.6f} seconds.")